### PR TITLE
Add Transaction Hash only to TxResult

### PR DESCRIPTION
**Pull request by patiee**

### CODE REVIEW

**Review by lavabyrd — APPROVED** [review]

File: codec.proto [hunk 1]
```diff
@@ -85,12 +85,7 @@ message ConsensusParams {
   VersionParams   version   = 4 [(gogoproto.nullable) = false];
 }
 
-message Data { repeated Tx txs = 1; }
-
-message Tx {
-  bytes tx_hash = 1;
-  bytes tx_raw  = 2;
-}
+message Data { repeated bytes txs = 1; }
```
Comment (sosedoff) [inline codec.proto:88]: Do we even need this any more?

File: codec.proto [hunk 2]
```diff
@@ -231,10 +226,11 @@ message Timestamp {
 }
 
 message TxResult {
-  uint64            height = 1;
-  uint32            index  = 2;
-  Tx                tx     = 3;
-  ResponseDeliverTx result = 4;
+  bytes             hash   = 1;
```
Comment (pienkowb) [inline codec.proto:229]: As this is an additional field we added, I would put this as the last field (`= 5`). That way if we receive a native TxResult, it should still decode properly.

File: codec.proto (hunk 1 above)
Comment (pienkowb) [reply codec.proto:88]: Are you suggesting that we get rid of the transaction payload from `EventBlock` and keep it only in `EventTx`?

File: codec.proto (hunk 2 above)
Comment (patiee) [reply codec.proto:229]: okay, good idea

**Pull request merged**